std/experimental/allocator/mallocator.d

```diff
@@ -0,0 +1,226 @@
+module std.experimental.allocator.mallocator;
+import std.experimental.allocator.common;
+
+/**
+   The C heap allocator.
+ */
+struct Mallocator
+{
+    unittest { testAllocator!(() => Mallocator.instance); }
+
+    /**
+    The alignment is a static constant equal to $(D platformAlignment), which
+    ensures proper alignment for any D data type.
+    */
+    enum uint alignment = platformAlignment;
+
+    /**
+    Standard allocator methods per the semantics defined above. The
+    $(D deallocate) and $(D reallocate) methods are $(D @system) because they
+    may move memory around, leaving dangling pointers in user code. Somewhat
+    paradoxically, $(D malloc) is $(D @safe) but that's only useful to safe
+    programs that can afford to leak memory allocated.
+    */
+    @trusted void[] allocate(size_t bytes) shared
+    {
+        import core.stdc.stdlib : malloc;
+        if (!bytes) return null;
+        auto p = malloc(bytes);
+        return p ? p[0 .. bytes] : null;
+    }
+
+    /// Ditto
+    @system bool deallocate(void[] b) shared
+    {
+        import core.stdc.stdlib : free;
+        free(b.ptr);
+        return true;
+    }
+
+    /// Ditto
+    @system bool reallocate(ref void[] b, size_t s) shared
+    {
+        import core.stdc.stdlib : realloc;
+        if (!s)
+        {
+            // fuzzy area in the C standard, see http://goo.gl/ZpWeSE
+            // so just deallocate and nullify the pointer
+            deallocate(b);
+            b = null;
+            return true;
+        }
+        auto p = cast(ubyte*) realloc(b.ptr, s);
+        if (!p) return false;
+        b = p[0 .. s];
+        return true;
+    }
+
+    /**
+    Returns the global instance of this allocator type. The C heap allocator is
+    thread-safe, therefore all of its methods and `it` itself are
+    $(D shared).
+    */
+    static shared Mallocator instance;
+}
+
+///
+unittest
+{
+    auto buffer = Mallocator.instance.allocate(1024 * 1024 * 4);
+    scope(exit) Mallocator.instance.deallocate(buffer);
+    //...
+}
+
+unittest
+{
+    static void test(A)()
+    {
+        int* p = null;
+        p = new int;
+        *p = 42;
+        assert(*p == 42);
+    }
+
+    test!Mallocator();
+}
+
+unittest
+{
+    static void test(A)()
+    {
+        Object p = null;
+        p = new Object;
+        assert(p !is null);
+    }
+
+    test!Mallocator();
+}
+
+version (Posix) private extern(C) int posix_memalign(void**, size_t, size_t);
+version (Windows)
+{
+    private extern(C) void* _aligned_malloc(size_t, size_t);
```

Comment:
I can propose the 3 functions (solution 2, [**cf this GIST**](https://gist.github.com/BBasile/be474f1621884734d7fc)) because actually there's a well known implementation that can be found easily on the web.

Where should i propose a PR. **druntime** or directly inside **mallocator.d** ?
I wait more comment about the problem.